table.cpp

```diff
@@ -95,11 +94,16 @@ TableCell::TableCell(float d, const std::string& c, int precision, const std::st
   academic_integrity = ai;
   event = e;
 
-  if (reason != "") {
-    hoverText = "class=\"hoverable-cell\" data-hover-text=\""+userName+" received a "+std::to_string(daysExtended)+" day extension due to "+reason+" on "+gID+"\" ";
-  } else {
-    hoverText = "class=\"hoverable-cell\" data-hover-text=\""+userName+" received a "+std::to_string(daysExtended)+" day extension without specified reason on "+gID+"\" ";
-  }
+  if (event == "Extension" && reason != "") {
+      extension = true;
+      inquiry = bad_status = override = version_conflict = cancelled = false;
+      hoverText = "class=\"hoverable-cell\" data-hover-text=\"" + userName + " received a " + std::to_string(daysExtended) + " day extension due to " + reason + " on " + gID + "\" ";
+  } else if (event == "Bad") {
+      bad_status = true;
+      override = inquiry = extension = version_conflict = cancelled = false;
+      hoverText = "class=\"hoverable-cell\" data-hover-text=\"" + userName + " received a bad status on " + gID + "\" ";
+      }
+  
```

Comment:
Is there a reason you are adding this in a separate if statement? Right below this code snippet we have existing if statements that are identical to your addition. I would say it is better to put it in there instead of duplicating the code.

Reply:
Fixed in [07ac6de](https://github.com/Submitty/RainbowGrades/pull/77/commits/07ac6dea469083e1954b9544df5eb49deea68582)